### PR TITLE
[tests] Regression test for #1956. [debugging] Logging improvements.

### PR DESCRIPTION
**Pull request by atamazov**
- Regression test for #1956
- By-products:
  - Logging improvements
    - [debug][MIOpenDriver] Quiet logging during computation of reference on GPU
    - [debug][tests] Always enable naive solvers and quiet logging for computation of reference on GPU
    - [debug] Improve logging in *GetWorkspaceSize()
  - [NFC] Fix build warnings in conv2d/3d_bias (unused variable)

### CODE REVIEW

**Comment by atamazov** [comment]
@junliume @johnny-keker Can you please restart CI, thanks!

File: src/solver/gemm_bwd.cpp [hunk 1]
```diff
@@ -641,6 +642,13 @@ bool GemmBwdRest::IsApplicable(const ExecutionContext& context,
     if(!GemmBwdBase::IsApplicable(context, problem))
         return false;
 
+#if WORKAROUND_ISSUE_1956
+    // Col2Im3dGPU has issues with address computations when workspace size > 4GiB.
+    if(problem.GetConv().GetSpatialDimension() == 3 &&
+       GetWorkspaceSize(context, problem) > 0xffffffffULL)
+        return false;
+#endif
```
Comment (atamazov) [inline src/solver/gemm_bwd.cpp]: ℹ️ This is the W/A. Other changes relate to by-products (logging improvements etc).

Comment (junliume) [reply src/solver/gemm_bwd.cpp]: Yep and we need to remove this WA when fully implemented

**Review by junliume — DISMISSED** [review]

File: src/solver/gemm_bwd.cpp (hunk 1 above)
Comment (atamazov) [reply src/solver/gemm_bwd.cpp]: @junliume Of course and that is the reason why we use tagging like WORKAROUND_ISSUE_1956 ))) Thanks for reviewing!

Comment (atamazov) [reply src/solver/gemm_bwd.cpp]: @junliume BTW theoretically the reason could still be a bug in the OCL compiler. If this is the case, then we'll make the value of WORKAROUND_ISSUE_1956 dependent on ROCm version.

**Comment by atamazov** [comment]
@junliume There is a full-blown fix in #1966. I do not know its ETA (and didn't tested it yet).

We can merge this as is and then remove WORKAROUND_ISSUE_1956 in #1966 or, alternatively, just close this PR and wait for #1966 to be ready. Decision is up to you.

**Comment by atamazov** [comment]
@junliume Update: Added regression test. Let's merge this as is and then remove WORKAROUND_ISSUE_1956 in #1966.

/cc @zjing14

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -2035,12 +2035,22 @@ add_custom_test(smoke_conv_solver_ConvWinograd3x3MultipassWrW GFX90A_DISABLED HA
...
-# REGRESSION TEST for issue #894.
+# Issue #894.
 # Can't be enabled for GFX10 due to WORKAROUND_SWDEV_271887
 COMMAND	MIOPEN_FIND_MODE=normal MIOPEN_DEBUG_FIND_ONLY_SOLVER=ConvOclDirectFwd1x1 $<TARGET_FILE:test_conv2d> ${MIOPEN_TEST_FLOAT_ARG} --verbose --disable-backward-data --disable-backward-weights --disable-verification-cache
     --cmode conv --pmode default --group-count 1 --input 1 16 7 7 --weights 16 16 1 1 --pads_strides_dilations 0 0 1 1 1 1
 )
 
+add_custom_test(test_regression_half_vega SKIP_UNLESS_ALL FLOAT_DISABLED HALF_ENABLED GFX908_DISABLED GFX90A_DISABLED
+# Issue #1956.
+COMMAND	MIOPEN_FIND_MODE=normal MIOPEN_DEBUG_FIND_ONLY_SOLVER='GemmBwdRest\\\\\\\;ConvDirectNaiveConvBwd' $<TARGET_FILE:test_conv3d> ${TEST_CONV_VERBOSE_B}
```
Comment (atamazov) [inline test/CMakeLists.txt]: [Note] `\\\\\\\;` is to escape `;`. CMake interprets this list three times. At the first time, `\\\\\\\;` is converted to `\\\;`, at the second time it becomes `\;`. Finally, `;` is inserted into build.make.

**Comment by junliume** [comment]
@atamazov can we close this one since #1966 is merged?

**Comment by atamazov** [comment]
@junliume NO, as it contains the **_regression test_** for #1956 + some useful enhancements. Let me update the PR.

[Attribution] New labels: https://github.com/ROCmSoftwarePlatform/MIOpen/labels/enhancement https://github.com/ROCmSoftwarePlatform/MIOpen/labels/testing https://github.com/ROCmSoftwarePlatform/MIOpen/labels/debugging https://github.com/ROCmSoftwarePlatform/MIOpen/labels/urgency_normal

**Comment by atamazov** [comment]
@junliume @zjing14 Ready for review & CI testing!

**Review by junliume — APPROVED** [review]
Changes look good to me, I need to restart CI when the nodes are back online.

**Pull request merged**